### ISSUE TITLE
Reliable way to test for H version.

### ISSUE THREAD

**Issue by anonymous1184** [thqby/AutoHotkey_H#72]
TL;DR: Feature request to add a built-in global to check for the H version. It can be any name, for example:

```ahk
MsgBox IsSet(A_IsH)
```

Personally, I only work with H because it provides stuff that I need and like (JSON5, Decimal, MT, etc.), but sometimes code portability is needed.

`#Requires` is out of the question as it will force just the H version.

Testing for the availability of features is a very good option:

```ahk
m := (UMap ?? Map)()
```

But there are times when testing for the version in order to decide what to use is needed:

```ahk
Test() {
    if (IsH()) {
        ; Use AHK_H fetures
    } else {
        ; Fallback to AHK_L features
    }
}
```

Currently, I use this:

```ahk
IsH() {
    if (!IsSet(isH)) {
        vi := Buffer()
        vi.Size := DllCall("version\GetFileVersionInfoSize", "Str", A_AhkPath, "UInt", 0)
        DllCall("version\GetFileVersionInfo", "Str", A_AhkPath, "UInt", 0, "UInt", vi.Size, "Ptr", vi)
        DllCall("version\VerQueryValue", "Ptr", vi, "Str", "\StringFileInfo\040904B0\ProductName", "Ptr*", &pInfo := 0, "UInt", 0)
        static isH := (StrGet(pInfo) = "AutoHotkey_H")
    }
    return isH
}
```

And works just fine, however, it relies on the binary to have the information set. That can change in any build; I don't recall exactly when, but a few years back, a v1.1\* was released without any version information (`FileGetVersion` command failed).

<sup>\* *Speaking of v1.1, binaries have [more version information][1] available.*</sup>

[1]: https://github.com/thqby/AutoHotkey_H/assets/53758552/79ca93aa-ec1d-4c2b-a5c0-c65193e30905

I could test for H-specific features, then again if those features were to be ported, the test would fail. That's why I no longer use testing for a few `A_*` vars unique to H:

```ahk
IsH() => (IsSet(A_ModuleHandle) && IsSet(A_ScriptStruct) && IsSet(A_GlobalStruct))
```

If a purpose-made variable is available, then there won't be any problem at all. I hope you take this into consideration.

As always, thank you very much for all the effort you put into your projects, and all the time you take to address issues.

**Comment by thqby** [comment]
IsH() {
    static _ := !!DllCall('GetProcAddress', 'ptr', DllCall('GetModuleHandleW', 'ptr', 0, 'ptr'), 'astr', 'NewThread', 'ptr')
    return _
}

**Comment by anonymous1184** [comment]
I (obviously) didn't think of that, AHK_L won't ever have an MT module (for the foreseeable future).

Thanks again!

**Comment by thqby** [comment]
what is MT?
This is a dll/exe export function.

**Comment by anonymous1184** [comment]
MT = Multi Thread, I meant that this return the entry point of the export/module.
